### PR TITLE
Docs for preferring RenderLogEvent on Target

### PR DESCRIPTION
**Pull request by 304NotModified**

### CODE REVIEW

File: src/NLog/Layouts/Layout.cs [hunk 1]
```diff
@@ -188,6 +188,8 @@ public virtual void Precalculate(LogEventInfo logEvent)
 
         /// <summary>
         /// Renders formatted output using the log event as context.
+        ///
+        /// Note: inside a <see cref="Targets.Target"/>, <see cref="Targets.Target.RenderLogEvent"/> is preferred for performance reasons.
```
Comment (snakefoot) [inline src/NLog/Layouts/Layout.cs:192]: Maybe have this as `<remark>` ?

Comment (304NotModified) [reply src/NLog/Layouts/Layout.cs]: done!

**Comment by snakefoot** [comment]
Another nice side-effect of using Target.RenderLogEvent is that it doesn't generate NullReferenceException when no Layout assigned.

File: src/NLog/Layouts/Typed/ValueTypeLayoutInfo.cs [hunk 2]
```diff
@@ -173,6 +174,7 @@ public CultureInfo ValueParseCulture
         /// <summary>
         /// Render Result Value
         /// </summary>
+        /// <remarks>Inside a <see cref="Target"/>, <see cref="Target.RenderLogEvent"/> is preferred for performance reasons.</remarks>
```
Comment (snakefoot) [inline src/NLog/Layouts/Typed/ValueTypeLayoutInfo.cs:177]: `RenderValue` on `ValueTypeLayoutInfo` is special, as it allows one to configure a Typed-Layout from the configuration-file and also has support for raw-values.

Comment (304NotModified) [reply src/NLog/Layouts/Typed/ValueTypeLayoutInfo.cs]: reverted this change

**Review by snakefoot — APPROVED** [review]
Looks good

**Pull request merged**